src/PIL/IcoImagePlugin.py

```diff
@@ -347,6 +348,85 @@ def load_seek(self):
         pass
 
 
+def _ico1_accept(prefix):
+    return prefix[0] == 1 and prefix[1] in (0, 1, 2)
+
+
+class Ico1ImageFile(ImageFile.ImageFile):
+    format = "ICO1"
+    format_description = "Windows 1.0 Icon"
+
+    def _open(self):
+        if not _ico1_accept(self.fp.read(2)):
+            msg = "not an ICO1 file"
+            raise SyntaxError(msg)
+
+        self.fp.seek(4, os.SEEK_CUR)
+        width = i16(self.fp.read(2))
+        height = i16(self.fp.read(2))
+        self._size = (width, height)
+        self._mode = "LA"
+
+        self.tile = [("ico1", (0, 0) + self.size, 14, None)]
+
+
+class Ico1Decoder(ImageFile.PyDecoder):
+    _pulls_fd = True
+
+    def decode(self, buffer):
+        data = bytearray()
+        bitmapLength = self.state.xsize * self.state.ysize // 8
+        firstBitmap = self.fd.read(bitmapLength)
+        secondBitmap = self.fd.read(bitmapLength)
+        for i, byte in enumerate(firstBitmap):
+            secondByte = byte ^ secondBitmap[i]
+            for j in reversed(range(8)):
+                first = byte >> j & 1
+                second = secondByte >> j & 1
+                data += b"\x00" if (first == second) else b"\xff"
+                data += b"\x00" if first else b"\xff"
+        self.set_as_raw(bytes(data))
+        return -1, 0
+
+
+class Ico1Encoder(ImageFile.PyEncoder):
+    _pushes_fd = True
+
+    def encode(self, bufsize):
+        firstBitmap = bytearray()
+        secondBitmap = bytearray()
+        w, h = self.im.size
+        for y in range(h):
+            for x in range(w):
+                l, a = self.im.getpixel((x, y))
+                if x % 8 == 0:
+                    firstBitmap += b"\x00"
+                    secondBitmap += b"\xff"
+                if not a:
+                    firstBitmap[-1] ^= 1 << (7 - x % 8)
+                if not l:
+                    secondBitmap[-1] ^= 1 << (7 - x % 8)
+        data = firstBitmap + secondBitmap
```

Comment:
`snake_case`? Also, `l` and `a` are somewhat opaque to me.

Reply:
I've pushed commits to update the variable names. It is an LA image being saved, so `l` is the L channel, and `a` the A channel.

Reply:
So the master branch is already patched, or we must wait?

Reply:
If I understand you correctly, this PR has not been merged into Pillow's main branch, no. It is on hold because there have been [concerns raised about whether this really needs to be part of Pillow](https://github.com/python-pillow/Pillow/issues/7320#issuecomment-1741270231).

Reply:
Isn't it possible to make it a plugin? Are plugins easy to install?

Reply:
If you just want to add it your own code, here is a separate plugin derived from this PR - [Ico1ImagePlugin.py.zip](https://github.com/python-pillow/Pillow/files/13358252/Ico1ImagePlugin.py.zip)

Unzip it, and then you can
```python
from PIL import Image
import Ico1ImagePlugin

im = Image.open("ico1both.ico")
```

Reply:
I think this would be useful in Pillow – people aren't going to find external plugins when they need them, so they'll end up writing the same code again, and that's no fun for anyone...

Reply:
I think we've merged new features for less good reasons so I'd like to see this go in. Worst case it comes out again in the future? 🤷